src/OrleansGetEventStore/GetEventStoreProvider.cs

```diff
@@ -0,0 +1,304 @@
+            var hostPort = int.Parse(connectionStringParts[1]);
+            var hostAddress = Dns.GetHostAddresses(hostName).First(a => a.AddressFamily == AddressFamily.InterNetwork);
+            
+            Log.Info($"Init Name={Name} ConnectionString={connectionString} Severity={Log.SeverityLevel}");
+
+            var username = config.GetProperty(UsernameParameterName, "admin");
+            var password = config.GetProperty(PasswordParameterName, "changeit");
+
+            var settings = ConnectionSettings.Create()
+                .KeepReconnecting().KeepRetrying()
+                .SetDefaultUserCredentials(new UserCredentials(username, password));
+
+            this.Connection = EventStoreConnection.Create(settings, new IPEndPoint(hostAddress, hostPort));
+            await this.Connection.ConnectAsync();
+
+            Log.Verbose("Successfully Connected");
+        }
+
+        public Task Close()
+        {
+            this.Connection.Close();
+            return TaskDone.Done;
+        }
+ 
+        private static readonly Dictionary<string,object> emptyCommitHeaders = new Dictionary<string, object>();
+
+        public async Task<bool> Append(string streamName, IEnumerable<KeyValuePair<Guid,object>> events, int? expectedVersion)
+        {
+            var eventsToSave = events.Select(e => ToEventData(e.Key, e.Value, emptyCommitHeaders)).ToArray();
+
+            if (eventsToSave.Length == 0)
+                return true;
+
+            var version = expectedVersion.HasValue
+                ? expectedVersion.Value == 0
+                    ? ExpectedVersion.NoStream
+                    : expectedVersion.Value - 1
+                : ExpectedVersion.Any;
+
+            if (eventsToSave.Length < pageSize)
+            {
+                try
+                {
+                    await this.Connection.AppendToStreamAsync(streamName, version, eventsToSave);
+                }
+                catch (WrongExpectedVersionException)
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                try
+                {
+                    var transaction = await this.Connection.StartTransactionAsync(streamName, version);
+
+                    var position = 0;
+                    while (position < eventsToSave.Length)
+                    {
+                        var pageEvents = eventsToSave.Skip(position).Take(pageSize);
+                        await transaction.WriteAsync(pageEvents);
+                        position += pageSize;
+                    }
+
+                    await transaction.CommitAsync();
+                }
+                catch (WrongExpectedVersionException)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static List<KeyValuePair<Guid, object>> emptyList = new List<KeyValuePair<Guid, object>>();
```

Comment:
Not tampering on purpose, but you are passing this collection around, and someone could potentially think it's fine to mutate it because it might have been created for him (the most common case). But since this is a cached empty collection, then that's not the case.